src/Http/RouteCollection.php

```diff
@@ -88,7 +88,23 @@ protected function fixPathPart(&$part, $key, array $parameters)
     public function getPath($name, array $parameters = [])
     {
         if (isset($this->reverse[$name])) {
+            $curr_max_count = 0;
             $parts = $this->reverse[$name][0];
+
+            // For a given route name, we want to choose the option that best matches the given parameters.
+            // Each routing option is an array of parts. Each part is either a constant string
+            // (which we don't care about here), or an array where the first element is the parameter name
+            // and the second element is a regex into which the parameter value is inserted, if the parameter matches.
+            foreach ($this->reverse[$name] as $parts_option) {
+                for ($i = 0; $i < count($parts_option); $i++) {
+                    $part = $parts_option[$i];
+                    if (is_array($part) && in_array($part[0], array_keys($parameters)) && $i > $curr_max_count) {
```

Comment:
Maybe replace `in_array($part[0], array_keys($parameters))` with `array_key_exists($part[0], $parameters))` or even the shorter `Arr::exists($parameters, $part[0])` ?